### PR TITLE
fix: PreviewPhrase becomes invalid when adding second a phrase (M2-8199) 

### PR DESCRIPTION
**Pull request by AlejandroCoronadoN**
### 📝 Description
The "Preview Phrase" button of the first phrase becomes disabled at the screen if second phrase is added

🔗 [Jira Ticket M2-8199](https://mindlogger.atlassian.net/jira/software/c/projects/M2/boards/4?assignee=712020%3A6322a990-2eaf-42f4-9fed-30af94baf20e&selectedIssue=M2-8199)

Changes include:

- Importing PhrasalTemplateResponseValuePhraseSchema on PhrasalTemplatePhrase to validate form input to display PhrasalPreview button.
- Deleted the isValid and isSubmmiting logic attached to the PhrasalPreview button and substitute it with the PhrasalTemplateResponseValuePhraseSchema validation. 

### 📸 Screenshots

image

### 🪤 Peer Testing

* Create an Applet
* Create at least one activity that is not a Phrase Builder.
* Create a new Phrase builder actyivity. 
* Complete the first phrase by filling the text, select the forst activitry you created as input for the second field and finally create a text for the last entry inside the Phrase. 
* At this point you should be able to click on the Preview Phrase button.
* Click on Add Phrase to insert a new Phrase
* The first Preview Phrase button should still by enabled and it should display the preview phrase. 

### ✏️ Notes
I making the validation using the PhrasalTemplateResponseValuePhraseSchema but and I'm passing field to the PhrasalTemplatePhrase to validate the full object. However I could have just validated the Array contained in this object that is already called by useFieldArray inside PhrasalTemplatePhrase (Avoiding passing field to the script in the first place)  Notice that we have an structure like fields.field.fields where fields.field is a PhrasalTemplateResponseValuePhraseSchema and fields.field.fields is a PhrasalTemplateResponseValuePhraseFieldSchema. 

`  const { fields, append, remove, swap } = useFieldArray({
    control: control as unknown as Control<{ values: TPhrasalTemplateField[] }>,
    name: phraseFieldsName as 'values',
  });`

### CODE REVIEW

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (farmerpaul) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: Nit: Long relative paths involving many consecutive `../` are generally discouraged. An absolute import in this case is easier to read:
```suggestion
import { PhrasalTemplateResponseValuePhraseSchema } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.schema';
```

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (farmerpaul) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: Nit: There's an ESLint warning about an unused variable, maybe when debugging your code earlier? Try to notice when you introduce ESLint warnings. (Tip: Whenever I create a PR, I usually review my PR's **Files changed** tab in GitHub to do a sanity check and catch debugging code that I meant to delete, because it happens a lot!)

It's true there are many existing ESLint warnings still in the codebase, but we are actively working to eliminate them as we encounter them. At the minimum, avoid introducing new ones. 😇🙏

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (farmerpaul) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: Curious why you thought it necessary to include `abortEarly: false`? The default is `true` because it's more performant. I ran some tests locally and it seems to work fine using the default behaviour, but maybe I missed an edge case?
```suggestion
      await PhrasalTemplateResponseValuePhraseSchema.validate(field);
```

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (farmerpaul) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: Nit: I think it's redundant to include both `formState` and `formState.isSubmitting` as deps here; changes to `formState.isSubmitting` is implied by changes to `formState`.
```suggestion
  }, [field, formState]);
```

**Review by farmerpaul — COMMENTED** [review]
Great work @AlejandroCoronadoN! Nice and simple, straightforward solution. 💪🏻 The only comments I had really were nitpicks, nothing major, but I'll leave it you to respond before I approve.

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (farmerpaul) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: The logic here seems to be backwards. A function that provides a validation result should be `true` if it's valid, `false` if invalid. I'll make a tweak to this to make the logic a little more intuitive.

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (farmerpaul) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: This inline comment doesn't seem to be applicable to the line that follows it; I'll move it somewhere more meaningful.

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (mbanting) [reply src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: Looks like this one still is unresolved and wasn't made as part of @AlejandroCoronadoN last [commit](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1977/commits/f6a19795dd1606b98a7498bfa4033abaabb9456c)

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (mbanting) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: @farmerpaul this will also need to be changed to 
```suggestion
      setPreviewPhraseDisabled(!result);
```
after updating `validatePhraseField` although I'm sure you know this already :)

**Review by mbanting — APPROVED** [review]
Approving with the remaining changes made by @farmerpaul. Thank you @AlejandroCoronadoN and @farmerpaul !

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx
Comment (farmerpaul) [reply src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx]: I've just taken care of this :+1:

**Pull request merged**
